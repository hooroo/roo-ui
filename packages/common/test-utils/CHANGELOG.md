# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
