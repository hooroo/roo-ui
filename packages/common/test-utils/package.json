{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
