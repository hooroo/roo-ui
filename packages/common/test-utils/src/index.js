export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
