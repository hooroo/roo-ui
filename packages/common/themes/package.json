{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
