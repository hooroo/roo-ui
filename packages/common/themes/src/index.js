import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
