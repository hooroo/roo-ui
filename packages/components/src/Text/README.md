# Text

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Text } from '@roo-ui/components';

export default (
  <Fragment>
    <Text>Generic Text</Text>
    <Text.paragraph>Text paragraph</Text.paragraph>
  </Fragment>
);
```
```

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [color](https://github.com/jxnblk/styled-system#color-responsive), [font size](https://github.com/jxnblk/styled-system#fontsize-responsive), [space](https://github.com/jxnblk/styled-system#space-responsive), or [typography](https://github.com/jxnblk/styled-system#typography).
