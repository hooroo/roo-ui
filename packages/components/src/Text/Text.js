import styled from 'styled-components';
import tag from 'clean-tag';
import {
  textStyle,
  color,
  fontSize,
  fontWeight,
  letterSpacing,
  lineHeight,
  space,
  textAlign,
  themeGet,
} from 'styled-system';

const Text = styled(tag.p)`
  margin: 0;
  margin-bottom: ${themeGet('space.2')};

  ${textStyle}
  ${color}
  ${fontSize}
  ${fontWeight}
  ${letterSpacing}
  ${lineHeight}
  ${space}
  ${textAlign}
`;

Text.propTypes = {
  ...textStyle.propTypes,
  ...color.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...letterSpacing.propTypes,
  ...lineHeight.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
};

Text.defaultProps = {
  textStyle: 'text',
};

Text.displayName = 'Text';

export default Text;
