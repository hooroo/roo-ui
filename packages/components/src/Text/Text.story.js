import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import Text from '.';
import README from './README.md';

storiesOf('Components|Text', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Fragment>
      <Text>
        Generic text.
      </Text>
      <Text.paragraph>
        A paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eget eros
        malesuada, suscipit turpis id, vehicula lorem. Etiam iaculis mi quis risus
        mattis posuere. Cras ultricies metus mauris, a dapibus lacus facilisis
        iaculis.
      </Text.paragraph>
    </Fragment>
  ));
