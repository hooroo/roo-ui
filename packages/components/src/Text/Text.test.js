import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import Text from '.';

describe('<Text />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Text>Hello world</Text>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
