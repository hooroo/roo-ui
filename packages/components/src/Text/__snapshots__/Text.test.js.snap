// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5;
  font-weight: 500;
}

<Clean.span
  className="c0"
  hidden={false}
  textStyle="text"
>
  Hello world
</Clean.span>
`;
